### PR TITLE
Use Echo middleware rather than depend on Echo's bundled JWT middleware

### PR DESCRIPTION
**Pull request by bittrance**
There are two reasons for this: 
1. The packaged middleware tends to return 400 rather than more correct 4XX codes
2. The second reason is that this switch helps #252.

### CODE REVIEW

File: oidcechojwt/echo_jwt_test.go [hunk 1]
```diff
@@ -115,13 +112,13 @@ func (h *testHandler) NewHandlerFn(claimsValidationFn options.ClaimsValidationFn
 func (h *testHandler) ToHandlerFn(parseToken oidc.ParseTokenFunc[oidctesting.TestClaims], opts ...options.Option) http.Handler {
 	h.tb.Helper()
 
-	echoParseToken := toEchoJWTParseTokenFunc(parseToken, opts...)
-	return testGetEchoRouter(h.tb, echoParseToken)
+	echoMiddleware := toEchoHandler(parseToken, opts...)
```
Comment (simongottschlag) [inline oidcechojwt/echo_jwt_test.go:115]: `oidcHandler`

File: oidcechojwt/echo_jwt_test.go [hunk 2]
```diff
@@ -115,13 +112,13 @@ func (h *testHandler) NewHandlerFn(claimsValidationFn options.ClaimsValidationFn
...
+	return testGetEchoRouter(h.tb, echoMiddleware)
 }
 
 func (h *testHandler) NewTestServer(opts ...options.Option) oidctesting.ServerTester {
 	h.tb.Helper()
 
-	echoParseToken := New[oidctesting.TestClaims](nil, opts...)
-	return newTestServer(h.tb, testGetEchoRouter(h.tb, echoParseToken))
+	echoMiddleware := New[oidctesting.TestClaims](nil, opts...)
```
Comment (simongottschlag) [inline oidcechojwt/echo_jwt_test.go:122]: `oidcHandler`

**Review by simongottschlag — CHANGES_REQUESTED** [review]
I like it! Would like the module to be renamed as well as the files to remove `jwt`. An update to the readme with the change would be good as well.

File: examples/shared/server_echo_jwt.go [hunk 3]
```diff
@@ -20,17 +18,14 @@ func newEchoJWTClaimsHandler[T any](c echo.Context) error {
 	return c.JSON(http.StatusOK, claims)
 }
 
-func RunEchoJWT[T any](parseToken echoJWTParseTokenFunc, address string, port int) error {
+func RunEchoJWT[T any](echoMiddleware echo.MiddlewareFunc, address string, port int) error {
```
[the remark this answers is not in the document]
Comment (bittrance) [reply examples/shared/server_echo_jwt.go:21]: jwt naming removed. 

It is not a handler, tho. That concept exists in Echo as well. A middleware in Echo is a handler factory. See example in https://echo.labstack.com/cookbook/middleware/ . How about `echo.OidcMiddleware`?

[the remark this answers is not in the document]
Comment (simongottschlag) [reply examples/shared/server_echo_jwt.go]: Yes, I'm fine with that!

[the remark this answers is not in the document]
Comment (bittrance) [reply examples/shared/server_echo_jwt.go:21]: Fixed.

File: oidcechojwt/echo_jwt_test.go [hunk 4]
```diff
@@ -28,19 +27,17 @@ func BenchmarkSuite(b *testing.B) {
 	oidctesting.RunBenchmarks(b, testName, newTestHandler(b))
 }
 
-func testGetEchoRouter(tb testing.TB, parseToken echoJWTParseTokenFunc) *echo.Echo {
+func testGetEchoRouter(tb testing.TB, echoMiddleware echo.MiddlewareFunc) *echo.Echo {
```
[the remark this answers is not in the document]
Comment (bittrance) [reply oidcechojwt/echo_jwt_test.go:30]: Fixed, see above.

File: oidcechojwt/echo_jwt_test.go (hunk 2 above)
Comment (bittrance) [reply oidcechojwt/echo_jwt_test.go:122]: Fixed, see above.

File: oidcechojwt/echo_jwt_test.go (hunk 1 above)
Comment (bittrance) [reply oidcechojwt/echo_jwt_test.go:115]: Fixed, see above.

**Review by simongottschlag — APPROVED** [review]

**Pull request merged**